src/core/mac/mac_frame.cpp

```diff
@@ -1048,6 +1070,140 @@ void TxFrame::ProcessTransmitAesCcm(const ExtAddress &aExtAddress)
 #endif // OPENTHREAD_RADIO
 }
 
+otError TxFrame::GenerateImmAck(const RxFrame &aFrame, bool aIsFramePending)
+{
+    uint16_t fcf = 0;
+
+    mChannel = aFrame.mChannel;
+    memset(&mInfo.mTxInfo, 0, sizeof(mInfo.mTxInfo));
+
+    fcf |= kFcfFrameAck;
+    if (aIsFramePending)
+    {
+        fcf |= kFcfFramePending;
+    }
+    fcf |= aFrame.GetVersion();
+    Encoding::LittleEndian::WriteUint16(fcf, mPsdu);
+
+    mPsdu[2] = aFrame.GetSequence();
+
+    mLength = kImmAckLength;
+
+    return OT_ERROR_NONE;
+}
+
+#if OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT
+otError TxFrame::GenerateEnhAck(const RxFrame &aFrame, bool aIsFramePending, const uint8_t *aIeData, uint8_t aIeLength)
+{
+    otError error = OT_ERROR_NONE;
+
+    uint16_t fcf = 0;
```

Comment:
Similar suggestion to start with `fcf  = kFcfFrameAck | kFcfFrameVersion2015 | kFcfSrcAddrNone `?